test/freenet/node/PeerLocationTest.java

```diff
@@ -0,0 +1,110 @@
+package freenet.node;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import junit.framework.TestCase;
+
+public class PeerLocationTest extends TestCase {
+
+    private static final double[][] PEER_LOCATIONS = new double[][] {
```

Comment:
I think that it's important to compare what the old logic was doing to the new one... maybe run it only when extensive tests are turned on... 

Have a look to smallestDistanceLegacy in
https://github.com/nextgens/fred/commit/f3f77b4e804581d6fe0204bc5c63283b740c2a28